src/caffe/test/test_gradient_based_solver.cpp

```diff
@@ -465,7 +465,7 @@ class GradientBasedSolverTest : public MultiDeviceTest<TypeParam> {
     for (int devices = 1; devices <= available_devices; ++devices) {
       // Configure batch size for single / multi device equivalence.
       // Constant data is needed for multi device as for accumulation.
-      num_ = kNum * devices;
+      num_ = kNum * devices * devices;
```

Comment:
We have kept going back in forth on this.  This enables multiGPU solver tests to work properly.  The reason this needs to be done is that num_ is a global and not passed through the tests and the solver tests read num to know how many participants they have.  With multiple solvers, this gets the correct batch size initialized.  There is likely a cleaner but more invasive way to do this.  Happy to hear about a better solution.